avm/ptn/lz/sub-vending/src/self/subResourceWrapper/deploy.bicep

```diff
@@ -0,0 +1,689 @@
+  for assignment in roleAssignmentsResourceGroupSelf: if (roleAssignmentEnabled && !empty(roleAssignmentsResourceGroupSelf)) {
+    dependsOn: [
+      createResourceGroupForLzNetworking
+    ]
+    name: take(
+      '${deploymentNames.createLzRoleAssignmentsRsgsSelf}-${uniqueString(assignment.principalId, assignment.definition, assignment.relativeScope)}',
+      64
+    )
+    params: {
+      location: virtualNetworkLocation
+      principalId: assignment.principalId
+      roleDefinitionIdOrName: assignment.definition
+      subscriptionId: subscriptionId
+      resourceGroupName: split(assignment.relativeScope, '/')[2]
+    }
+  }
+]
+
+module createLzRoleAssignmentsRsgsNotSelf '../../modules/Microsoft.Authorization/roleAssignments/deploy.bicep' = [
+  for assignment in roleAssignmentsResourceGroupNotSelf: if (roleAssignmentEnabled && !empty(roleAssignmentsResourceGroupNotSelf)) {
+    name: take(
+      '${deploymentNames.createLzRoleAssignmentsRsgsNotSelf}-${uniqueString(assignment.principalId, assignment.definition, assignment.relativeScope)}',
+      64
+    )
+    params: {
+      location: virtualNetworkLocation
+      principalId: assignment.principalId
+      roleDefinitionIdOrName: assignment.definition
+      subscriptionId: subscriptionId
+      resourceGroupName: split(assignment.relativeScope, '/')[2]
+    }
+  }
+]
+
+module createResourceGroupForDeploymentScript 'br/public:avm/res/resources/resource-group:0.2.3' = if (!empty(resourceProviders)) {
+  scope: subscription(subscriptionId)
+  name: deploymentNames.createResourceGroupForDeploymentScript
+  params: {
+    name: deploymentScriptResourceGroupName
+    location: deploymentScriptLocation
+    enableTelemetry: enableTelemetry
+  }
+}
+
+module createManagedIdentityForDeploymentScript 'br/public:avm/res/managed-identity/user-assigned-identity:0.1.0' = if (!empty(resourceProviders)) {
+  scope: resourceGroup(subscriptionId, deploymentScriptResourceGroupName)
+  name: deploymentNames.createDeploymentScriptManagedIdentity
+  dependsOn: [
+    createResourceGroupForDeploymentScript
+  ]
+  params: {
+    location: deploymentScriptLocation
+    name: deploymentScriptManagedIdentityName
+    enableTelemetry: enableTelemetry
+  }
+}
+
+module createRoleAssignmentsDeploymentScript '../../modules/Microsoft.Authorization/roleAssignments/deploy.bicep' = if (!empty(resourceProviders)) {
+  name: take('${deploymentNames.createRoleAssignmentsDeploymentScript}', 64)
+  params: {
+    location: deploymentScriptLocation
+    principalId: !empty(resourceProviders) ? createManagedIdentityForDeploymentScript.outputs.principalId : ''
+    roleDefinitionIdOrName: 'Contributor'
+    subscriptionId: subscriptionId
+  }
+}
+
+module createRoleAssignmentsDeploymentScriptStorageAccount '../../modules/Microsoft.Authorization/roleAssignments/deploy.bicep' = if (!empty(resourceProviders)) {
```

Comment:
Please note that there is now also a published pattern for this

Reply:
I've incorporated it and testing now :)